pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -910,18 +914,40 @@ func GetSelectorRegexMap(l7 policy.L7DataMap) (CachedSelectorREEntry, error) {
 		reStrings := make([]string, 0, len(l7Rules.DNS))
 		for _, dnsRule := range l7Rules.DNS {
 			if len(dnsRule.MatchName) > 0 {
-				dnsRuleName := strings.ToLower(dns.Fqdn(dnsRule.MatchName))
-				dnsPatternAsRE := matchpattern.ToRegexp(dnsRuleName)
-				reStrings = append(reStrings, "("+dnsPatternAsRE+")")
+				dnsRuleName := FromFqdn(dnsRule.MatchName)
+				dnsPatternAsRE, _ := CustomToRegexp(Reverse(dnsRuleName))
+				reStrings = append(reStrings, dnsPatternAsRE)
 			}
 			if len(dnsRule.MatchPattern) > 0 {
-				dnsPattern := matchpattern.Sanitize(dnsRule.MatchPattern)
-				dnsPatternAsRE := matchpattern.ToRegexp(dnsPattern)
-				reStrings = append(reStrings, "("+dnsPatternAsRE+")")
+				dnsPattern := FromFqdn(dnsRule.MatchPattern)
+				dnsPatternAsRE, isAllAllowed := CustomToRegexp(Reverse(dnsPattern))
+				// If the pattern is a wildcard, quit the regex calculation and insert a simple allow
+				// all rule. This will reduce memory usage and improve performance significantly when
+				// a wildcard rule is present together with a set of other rules.
+				// TODO do even need any validation at all here, since we will only ever get valid domains?
+				// eg do only "."
+				if isAllAllowed {
```

Comment:
```suggestion
				dnsPatternAsRE, isWildcard := CustomToRegexp(Reverse(dnsPattern))
				// If the pattern is a wildcard, quit the regex calculation and insert a simple allow
				// all rule. This will reduce memory usage and improve performance significantly when
				// a wildcard rule is present together with a set of other rules.
				// TODO do even need any validation at all here, since we will only ever get valid domains?
				// eg do only "."
				if isWildcard {
```

Reply:
Can you elaborate on the TODO?

> TODO do even need any validation at all here, since we will only ever get valid domains?

Reply:
Yeee. My thought is that in case we hit a wildcard, we don't really care about the regex to begin with. in that case, can we just return `^.*$` or something to always short circuit the regex?

And thanks for the suggestion, that looks better!